db/collection_index.go

```diff
@@ -197,9 +197,13 @@ func (c *collection) updateIndexedDoc(
 		return err
 	}
 	for _, index := range c.indexes {
-		err = index.Update(ctx, txn, oldDoc, doc)
-		if err != nil {
-			return err
+		// We only need to update the index if one of the indexed fields
+		// on the document has been changed.
+		if isUpdatingIndexedFields(index, oldDoc, doc) {
```

Comment:
question(cc: @islamaliev): Should this be done inside the `index.Update` func, instead of here? This will A) allow different index types to handle this differently B) Gurantees this behavior is consistent regardless of where `index.Update` is called from (vs doing it here which is only fixed for this particular call path).

Reply:
I also thought about this, but though it would be unlikely to have any other update behaviour.
Although, thinking about it again we don't know what will be in the future especially when we decide to index array or json fields or some other fancy stuff.

Reply:
Moved ✅